src/components/FormAddReservation.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { useEffect, useState } from 'react';
+import { PropTypes } from 'prop-types';
+import { useGetRoomsDetailsQuery, useCreateReservationMutation } from '../api/roomsData';
+import { useCurrentUserQuery } from '../api/authLog';
+
+const FormAddReservation = ({ roomId }) => {
+  const [user, setUser] = useState({});
+  const [username, setUsername] = useState('');
+  const [room, setRoom] = useState('');
+  const [date, setDate] = useState('');
+  const [city, setCity] = useState('');
+  const { data, error, isLoading } = useGetRoomsDetailsQuery();
+  const [createReservation] = useCreateReservationMutation();
+  const { data: currentUser } = useCurrentUserQuery();
+  const selectedOption = document.getElementById(roomId);
+
+  useEffect(() => {
+    setUser(currentUser);
+    if (selectedOption !== null) selectedOption.selected = true;
+  });
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+
+  if (error) {
+    return (
+      <p>
+        Error:
+        {error.message}
+      </p>
+    );
+  }
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const uid = user.id;
+    createReservation({
+      date,
+      city,
+      user_id: uid,
+      room_id: room,
+    }).unwrap()
+      .then((result) => {
+        console.log('Reservation created successfully:', result);
+      })
+      .catch((error) => {
+        console.error('Error creating reservation:', error);
+      });
+  };
+
```

Comment:
There seems to be an issue with the reservations page. On the first load of the page, I get this error:

![errorreserv](https://user-images.githubusercontent.com/111214189/235218707-58df1e75-4cd7-4dee-8d08-3798ecf0bb91.png)

After manually refreshing the page, the form appears and works as expected. Kindly check the way you are getting your username to avoid this issue 👍 This will help your app work flawlessly and give users a better experience.